modules/assim.sequential/R/sda.enkf.R

```diff
@@ -309,20 +314,14 @@ sda.enkf <- function(settings, obs.mean, obs.cov, IC = NULL, Q = NULL) {
     
   })
   
-    tobit2space.model <- nimbleCode({
-
-    for(n in 1:nens){
-        y.censored[n,1:N] ~ dmnorm(muf[1:N],pf[1:N,1:N])
-
-        for(i in 1:N){
-            y.ind[n,i] ~ dinterval(y.censored[n,i], interval[i,1:2])
-        }
+  tobit2space.model <- nimbleCode({
+    
+    y.censored[1:N] ~ dmnorm(muf[1:N], prec = pf[1:N,1:N])
+    
+    for(i in 1:N){
+      y.ind[i] ~ dconstraint(y.censored[i] > 0)
     }
-
-    #Priors
-    pf[1:N,1:N]  ~ dwish(aq[1:N,1:N],bq)
-    muf[1:N]  ~ dmnorm(mu.prior[1:N],cov.prior[1:N,1:N])
```

Comment:
Not following how this works now that the priors on muf and pf have been removed

Reply:
The purpose of this short model is to impute the means for the state variables that are zero inflated, so that they are in tobit space in the analysis model. I don't think it was correct to estimate them before.

Reply:
I understand the purpose of the model, what I don't understand is how a model that fits ```muf``` and ```pf``` can run without priors on those variables.

Reply:
They aren't fit. when y.censored = 0, y.censored is imputed with mean muf and covariance pf

Reply:
Ok, but then where does muf and pf come from? They aren't the sample mean and cov because those are zero-inflated and will give you the wrong value. I thought the whole point of this model was was to estimate those variables so we can pass them on to main filter model?

Reply:
I'm getting confused because if we estimate them we change the sample means and covariances for the state variables that are not zero inflated. At least that is what was happening before. I think we only want to impute the state variables that are all zeros and in those cases the mean is zero and we were picking the minimum covariance to be the stand in covariance. But, you're saying we want to estimate the whole muf and pf before passing it into the filter model and assume that all of the state variables are zero inflated?

Reply:
* I've always assumed that the thing we were passing into the filter model was muf and pf as a prior on X, not passing in the actual ensemble members. If you were doing the latter I don't see why we need the extra step of this model, since we could just do it all as one model.
* We cannot impute the state variables of the zeros without also estimating their mean and variance! Their sample mean and variance is still wrong. For example, if a state is >50% of it's observations at zero, then the true mean has to be negative, but the sample mean can never ever be negative. This is an extreme case, but the bias in mu will still be there in all cases and will just accumulate to be larger and larger the more zeros in the data.
* I agree that it's odd for non-zero bound state variables to have their means and variances shift away from the sample mean and variance based on their covariance with zero-bound state variables. But it seems like we should figure out why that's happening, and whether it's actually the correct answer or not. To me, this code doesn't feel like the correct solution.
* State variables where all or almost all ensemble members are zero definitely seems like a tough case. But in those cases the mean definitely isn't zero -- it's some unknown negative number. Problem is there's almost no information in the ensemble to figure out what that negative number SHOULD be. Seems like whatever the model comes up with would be pretty dependent upon the prior (which presumably would make -1 much more likely than -1000). Would LIKE to deal with these cases explicitly, but I could see needing a kluge/hack in place that excludes such terms from the state estimation and instead inserts a mean that's set to have high uncertainty, zero covariances, and a mean that results in negligible probability of positive values (the exact value might scale with ensemble size). We'd want this to be set so that the data can easily nudge the model, but not be so strong as to allow it to follow noise in the data (after all, the model did reach a consensus that the species should be extinct)